### ISSUE TITLE
Add custom labels to ServiceMonitor

### ISSUE THREAD

**Issue by DodgeCamaro** [alexandrevilain/temporal-operator#423]
For scrape metrics by prometheus-operator, ServiceMonitor needs to have labels like this:
```
release: <release name by kube-prometheus-stack helm chart>
```
And manifest something like this
```
apiVersion: temporal.io/v1beta1
kind: TemporalCluster
metadata:
  name: temporal
spec:
  version: 1.20.2
  numHistoryShards: 4
  metrics:
    enabled: true
    prometheus:
      listenPort: 9090
      scrapeConfig:
        serviceMonitor:
          enabled: true
          labels: 
            release: monitoring

```

**Comment by alexandrevilain** [comment]
Hi! 

Thanks for the suggestion, looks like a required feature!

**Comment by alexandrevilain** [comment]
Released on 0.14.0

**Comment by DodgeCamaro** [comment]
Works well, thanks!
